### ISSUE TITLE
TypeScript example

### ISSUE THREAD

**Issue by eestein** [danialfarid/ng-file-upload#1553]
Hi, I couldn't find an example using TS. Is there any? I've already installed the typings `typings install dt~ng-file-upload --save --global -SG` but I can't find the example on how it's used.

Any ideas?

Thanks!

**Comment by apiweb** [comment]
Please, add a Typescript example to use ng-file-upload.
It is almost impossible to find examples using ng-file-upload with typescript, an example would be very welcome.

**Comment by andersnorgaard** [comment]
What kinds of examples are you looking for @eestein  and @apiweb ? I use the examples with Typescript with minimal changes.

**Comment by eestein** [comment]
@andersnorgaard basic examples like a "Getting Started" tutorial. I wasn't able to get the lib to work using TS so an example would tell me what am I doing wrong.

**Comment by andersnorgaard** [comment]
@eestein But do you already have a working Typescript + AngularJS setup? I did, and adding this module was painless.

**Comment by stombeur** [comment]
When using this in typescript, if you inject the upload service as a typed variable, e.g. 
private Upload: angular.angularFileUpload.IUploadService,

then the config object needs to be typed as angular.angularFileUpload.IFileUploadConfigFile
and that does not have a file property.

Injecting the service as any and then just using the examples in JS works, but for me it's not clear how we can do this in a properly typed way.

**Comment by andersnorgaard** [comment]
@stombeur Yes. I just use the upload service as "any" also.

**Comment by arlevitt** [comment]
I agree... would it be possible to see an actual example of a typescript controller having this service injected into it?

**Comment by danialfarid** [comment]
For version 2 all the code is in TypeScript and I will eventually port it back to Angular 1 to be able to use the same code base. So there is the plan to have the code in TypeScript.

**Comment by ravishan110** [comment]
@andersnorgaard can you show how you used it in typescript. It will be very helpful for me and many others like me.

**Comment by sonip396** [comment]
I am also trying to implement the lib in a running TypeScript project and it's very difficult with no examples, it would be nice if somebody already implemented it and could share how to do it.

**Comment by andyrus** [comment]
Hi I am using angular 1.5 and typescript, is there an example anywhere on how this is implemented using typescript yet?  Many Thanks
